solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -472,33 +401,30 @@ library DKG {
             "result under challenge is different than the submitted one"
         );
 
-        // Compute the actual group members hash by selecting actual members IDs
-        // based on seed used for current DKG execution.
-        bytes32 actualGroupMembersHash = keccak256(
-            abi.encodePacked(
-                self.sortitionPool.selectGroup(groupSize, bytes32(self.seed))
-            )
-        );
+        try
+            self.dkgValidator.validate(result, self.seed, self.startBlock)
+        returns (bool isValid, string memory errorMsg) {
+            if (isValid) {
+                revert("unjustified challenge");
+            }
 
-        require(
-            keccak256(abi.encodePacked(result.members)) !=
-                actualGroupMembersHash,
-            "unjustified challenge"
-        );
+            emit DkgResultChallenged(
+                self.submittedResultHash,
+                msg.sender,
+                errorMsg
+            );
+        } catch {
+            // if the validation reverted we consider the DKG result as invalid
+            emit DkgResultChallenged(
+                self.submittedResultHash,
+                msg.sender,
+                "validation reverted"
+            );
+        }
```

Comment:
```
INFO:Detectors:
DKG.challengeResult(DKG.Data,DKG.Result).isValid (contracts/libraries/DKG.sol#406) is a local variable never initialized
DKG.challengeResult(DKG.Data,DKG.Result).errorMsg (contracts/libraries/DKG.sol#406) is a local variable never initialized
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#uninitialized-local-variables
INFO:Detectors:
DKG.challengeResult(DKG.Data,DKG.Result) (contracts/libraries/DKG.sol#383-439) ignores return value by self.dkgValidator.validate(result,self.seed,self.startBlock) (contracts/libraries/DKG.sol#404-423)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#unused-return
INFO:Detectors:
Variable 'DKG.challengeResult(DKG.Data,DKG.Result).isValid (contracts/libraries/DKG.sol#406)' in DKG.challengeResult(DKG.Data,DKG.Result) (contracts/libraries/DKG.sol#383-439) potentially used before declaration: isValid (contracts/libraries/DKG.sol#407)
Variable 'DKG.challengeResult(DKG.Data,DKG.Result).errorMsg (contracts/libraries/DKG.sol#406)' in DKG.challengeResult(DKG.Data,DKG.Result) (contracts/libraries/DKG.sol#383-439) potentially used before declaration: DkgResultChallenged(self.submittedResultHash,msg.sender,errorMsg) (contracts/libraries/DKG.sol#411-415)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#pre-declaration-usage-of-local-variables
```

I am confused about this warning. 🤔

Reply:
Ignored checks in ab1e64985bde809a60c4e3152ce50eba0d56efba and reported the problem: https://github.com/crytic/slither/issues/982.